libraries/Microsoft.Bot.Configuration/BotConfiguration.cs

```diff
@@ -240,21 +275,20 @@ public ConnectedService DisconnectServiceByNameOrId(string nameOrId)
         }
 
         /// <summary>
-        /// remove a service by type and id
+        /// remove a service by id.
         /// </summary>
-        /// <param name="type"></param>
-        /// <param name="id"></param>
-        public void DisconnectService(string type, string id)
+        /// <param name="id">id of the service</param>
+        public void DisconnectService(string id)
         {
-            var svs = new List<ConnectedService>(this.Services);
+            var svs = new List<ConnectedService>(Services);
 
             for (var i = 0; i < svs.Count(); i++)
```

Comment:
Given you are copying the List consider a Linq Where to eliminate the values you don't want in the process of copying.
(Actually are all these copies required?)